Move JsonIterator to NMT

iuravic opened the pull request:
Refactor from here to https://github.com/Automattic/newspack-migration-tools/pull/19
Also moved the following:
- `src/Utils/BatchLogic.php`, because it is used by `JsonInterator`
- `src/Utils/Logger.php`, because `JsonInterator` uses it

naxoc reviewed (commented):
Because the namespace changed in the NMT, we'd need to run a search and replace on `NewspackCustomContentMigrator\Utils\BatchLogic` -> `Newspack\MigrationTools\Util`.

We need to keep the logger class here. See the NMT PR

iuravic commented:
> Because the namespace changed in the NMT, we'd need to run a search and replace on `NewspackCustomContentMigrator\Utils\BatchLogic` -> `Newspack\MigrationTools\Util`.
> 
> We need to keep the logger class here. See the NMT PR

Ah, of course, how clumsy of me. Thank you @naxoc, now fixed in [221debb](https://github.com/Automattic/newspack-custom-content-migrator/pull/544/commits/221debbb4b7bef35afc5ee77b142b7e0464a67cc).

Also in [6624e97](https://github.com/Automattic/newspack-custom-content-migrator/pull/544/commits/6624e97b957d0c561e39cc713982860b2cd626a0) now updated usage of `Logger` from `NewspackCustomContentMigrator\Utils\Logger` to `Newspack\MigrationTools\Util\Logger`.

`JsonIterator` itself is only used by one archived migration, so no need to change any of that.

iuravic commented:
@naxoc , ready for re-check,
- brought back the `Logger` here [754b043](https://github.com/Automattic/newspack-custom-content-migrator/pull/544/commits/754b043bfa11573ae3d8b0b4482a5ef6c0c0ab2d)
- fixed new namespace for `BatchLogic` here [221debb](https://github.com/Automattic/newspack-custom-content-migrator/pull/544/commits/221debbb4b7bef35afc5ee77b142b7e0464a67cc)
